src/library/libraryview.cpp

```diff
@@ -709,6 +737,29 @@ void LibraryView::FilterReturnPressed() {
   emit doubleClicked(currentIndex());
 }
 
+void LibraryView::SearchForArtist() {
+  SaveFocus();
+  if (!last_selected_song_.albumartist().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.albumartist().simplified());
+  } else if (!last_selected_song_.artist().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.artist().simplified());
+  }
+}
+
+void LibraryView::SearchForAlbum() {
+  SaveFocus();
+  if (!last_selected_song_.album().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.album().simplified());
+  }
+}
+
+void LibraryView::SearchForThis() {
+  SaveFocus();
+  if (!last_selected_container_.isEmpty()) {
+    filter_->ShowInLibrary(last_selected_container_.simplified());
```

Comment:
Is there any way to get a more predictable result for last_selected_container_? Right now, it varies based upon the sorting/grouping method used. For example: 

![2chainz](https://cloud.githubusercontent.com/assets/144119/23931689/bc3b2f9c-0909-11e7-8ac9-88ff88eda9f1.png)

selecting "search for this" results in a search for `2013B.O.A.T.S. II #MeTime`. 

Searching for 
![screenshot from 2017-03-14 23-00-40](https://cloud.githubusercontent.com/assets/144119/23931743/10b35df6-090a-11e7-97ef-bd45e9260c98.png)

results in a search for `10302009 i  festival 8 ca`